packages/mask/src/extension/popups/pages/Personas/components/ProfileList/index.tsx

```diff
@@ -88,26 +119,113 @@ export const ProfileList = memo(() => {
         [],
     )
 
-    const [, onDisconnect] = useAsyncFn(async (identifier: ProfileIdentifier) =>
-        Services.Identity.detachProfile(identifier),
+    const onDisconnect = useCallback(
+        (identifier: ProfileIdentifier, is_valid?: boolean, platform?: NextIDPlatform, identity?: string) => {
+            if (is_valid) {
+                setUnbind({
+                    identifier,
+                    platform,
+                    identity,
+                })
+                return
+            }
+            Services.Identity.detachProfile(identifier)
+        },
+        [],
     )
 
+    const { value: mergedProfiles, retry: refreshProfileList } = useAsyncRetry(async () => {
+        if (!currentPersona) return []
+        const publicHexKey = await Services.Helper.queryPersonaHexPublicKey(currentPersona.identifier)
+        if (!publicHexKey) return currentPersona.linkedProfiles
+        const response = await queryExistedBindingByPersona(publicHexKey)
+        if (!response) return currentPersona.linkedProfiles
+
+        return currentPersona.linkedProfiles.map((profile) => {
+            const target = response.proofs.find(
+                (x) =>
+                    profile.identifier.userId.toLowerCase() === x.identity.toLowerCase() &&
+                    profile.identifier.network.replace('.com', '') === x.platform,
```

Comment:
Maybe we need a mapping for nextID platform to SNS network.

Reply:
It is possible to add static methods to the ProfileIdentifier class to get the platform. But I'm not sure if this is possible.